Add page documenting how to test other targets (#1236)
ferrocene/ferrocene: ferrocene/doc/internal-procedures/src/setup-local-env.rst

Required software
-----------------

To develop Ferrocene locally, you’ll need to have the software required to build
Rust, as well as the following:

* ``git``, as the version control system used by Ferrocene.

* ``uv``, to manage the environment used to build the documentation. `Installation
  instructions. <https://docs.astral.sh/uv/getting-started/installation/>`_

* ``AWS CLI v2``, version **2.9.0 or greater**, needed to interact with our AWS
  account. Note that most Linux distributions only include version 1 of the AWS
  CLI, while we explicitly require version 2.

:ref:`x86_64-unknown-linux-gnu`
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> On Ubuntu 24.10, the following commands can be used:

Reviewer: ```suggestion
On Ubuntu 24.10, install the software requirements by running:
```

Just less wishy washy language :-p

.. code-block:: bash

   sudo apt install ninja-build bzip2 cmake gcc g++ awscli
   curl -LsSf https://astral.sh/uv/install.sh | sh

.. note::

   On versions earlier than Ubuntu 24.10 the ``awscli`` package is not newer than
   2.9.0, and should not be used.

> On other distributions, you will likely need to adapt the command for the
> relevant package manager and any modified package names.

Reviewer: ```suggestion
To install the requirements on other distributions, adapt the command for the relevant package manager and any modified package names.
```

For example, on Arch Linux ``awscli`` can be obtained from the
`AUR <https://aur.archlinux.org/packages/aws-cli-v2>`_, and the remaining packages
from the official repositories:

.. code-block:: bash

   sudo pacman -S ninja bzip2 cmake gcc uv

:ref:`aarch64-apple-darwin`
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> If you don't already have XCode set up, get the XCode command line tools as

Reviewer: maybe s/guided/described/ ?
